addons/sale/controllers/product_configurator.py

```diff
@@ -56,18 +57,22 @@ def get_product_configurator_values(
             )
         if not combination:
             combination = product_template._get_first_possible_combination()
+        currency = request.env['res.currency'].browse(currency_id)
+        pricelist = request.env['product.pricelist'].browse(pricelist_id)
+        so_date = datetime.fromisoformat(so_date)
```

Comment:
Because these calls to `browse` were made for each product (main and optionals), even though the results were identical for all products. In theory, this didn't result in extra DB calls, but it still added avoidable duplication (and there was probably a negligible overhead due to cache lookups).